essay/swar/latency.md

```diff
@@ -107,24 +107,48 @@ Curiously enough, our SWAR library gives that capability directly; in the `strle
+In our Robin Hood Hash Table implementation there are many examples of our generation of a SIMD of boolean results, querying for the first "match" (or the least significant lane that matched) and the immediate use of those booleans as SIMD inputs for further computation. I believe this helps our SWAR, ["software SIMD"](https://github.com/thecppzoo/zoo/blob/em/essay-swar-latency.md/glossary.md#software-simd) implementations to be competitive with the software that uses "hardware SIMD".
 
-This also shows a notorious deficiency in the efforts to standardize in C++ the interfaces for hardware SIMD: The results of lane-wise processing (vertical processing) that generate booleans would generate "mask" values of types that are bitfields of the results of comparisons, an "scalar" to be used in [ the non-vector "General Purpose Registers", GPRs, not SIMD type values [^3].  So, if you were to consume these results in further SIMD processing, you'd need to convert and adjust the bitfields in general purpose registers into SIMD registers, this "back and forth SIMD-scalar" conversions would add further performance penalties that compiler optimizers might not be able to eliminate, for a variety of reasons that are out of the scope of this document. Wait for us to write an essay that contains the phrase "cuckoo-bananas".
+I think SIMD ISAs do not offer the way to identify the "first" or least significant lane that matched a condition, typically, one must convert a SIMD register that contains the result of the comparison in some way (typically all bits in the either set or reset) to a "mask" scalar value.
 
-Our SWAR library, in this regard, is "future proof": boolean results are just an specific form of SIMD values.  We use this already for SIMD operations such as saturated addition: unsigned overflow (carry) is detected lane wise and lane-wise propagated to the whole lane, without conversion to scalar.  Saturation is implemented seamlessly as after-processing of computation, without having to convert to "scalar" (the result mask) and back.
+This also shows a notorious deficiency in the efforts to standardize in C++ the interfaces for hardware SIMD: The results of lane-wise processing (vertical processing) that generate booleans would generate "mask" values of types that are bitfields of the results of comparisons, an "scalar" to be used in the non-vector "General Purpose Registers", GPRs, not SIMD type values [^3].  So, if you were to consume these results in further SIMD processing, you'd need to convert and adjust the bitfields in general purpose registers into SIMD registers, this "back and forth SIMD-scalar" conversions would add further performance penalties that compiler optimizers might not be able to eliminate, for a variety of reasons that are out of the scope of this document. Wait for us to write an essay that contains the phrase "cuckoo-bananas".
```

Comment:
s/an "scalar"/a "scalar"
s/this "back and forth SIMD-scalar" conversions/these "back and forth SIMD-scalar" conversions
" compiler optimizers might not be able to eliminate" feels like a WILD understatement. I have no idea how a compiler would eliminate a single one of these cases currently.